src/virtualenv/create/via_global_ref/_virtualenv.py

```diff
@@ -4,7 +4,10 @@
 import os
 import sys
 
-VIRTUALENV_PATCH_FILE = os.path.join(__file__)
+try:
+    VIRTUALENV_PATCH_FILE = os.path.join(__file__)
+except NameError:
+    VIRTUALENV_PATCH_FILE = None
```

Comment:
I'm not a big fan of try/catch; can we instead just if/else with __file__ is None?

Reply:
Neither am I but the issue is that `__file__` is undefined